### PR TITLE
Properly propagate DD4hep dependencies to downstream packages

### LINKED ISSUE #26: MarlinUtil linkage with DD4hep
## Issue

Compiling a custom `Marlin` processor and linking against `MarlinUtil` still does not allow of usage of `<<` operator for `dd4hep::rec::StructExtension` failing during the compilation with `undefined reference` error.

#### test.hpp
```
#include "marlin/Processor.h"
class Tester : public marlin::Processor {
    public:
        Tester();
        marlin::Processor* newProcessor() {return new Tester;}
        void init();
        void processEvent(LCEvent* evt){};
        void end(){};
};
```
#### test.cpp
```
#include "test.hpp"
#include "DD4hep/Detector.h"
#include "DDRec/DetectorData.h"

Tester aTester;

Tester::Tester() : Processor("Tester"){}

void Tester::init(){

    dd4hep::rec::FixedPadSizeTPCData data = dd4hep::rec::FixedPadSizeTPCData();
    // commenting the next line makes compilation work
    std::cout<<data.toString();
}
```
#### CMakeLists.txt
```
cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
project(Tester)

find_package(ILCUTIL REQUIRED COMPONENTS ILCSOFT_CMAKE_MODULES)
include(ilcsoft_default_settings)


include_directories(${PROJECT_SOURCE_DIR}/include)
add_library(${PROJECT_NAME} SHARED ${PROJECT_SOURCE_DIR}/src/test.cpp)


### DEPENDENCIES ###
find_package(Marlin REQUIRED)
find_package(MarlinUtil REQUIRED)

FOREACH(pkg Marlin MarlinUtil)
   include_directories(${${pkg}_INCLUDE_DIRS})
   target_link_libraries(${PROJECT_NAME} ${${pkg}_LIBRARIES})
ENDFOREACH()

install(TARGETS ${PROJECT_NAME} DESTINATION ${PROJECT_SOURCE_DIR}/lib)
```
#### Compile with
```
source /cvmfs/ilc.desy.de/sw/x86_64_gcc82_centos7/v02-02-02/init_ilcsoft.sh
mkdir build && cd build
cmake -C $ILCSOFT/ILCSoft.cmake ..
make
```

#### Output
```
CMakeFiles/Tester.dir/src/Tester.cpp.o: In function `Tester::init()':
/cvmfs/ilc.desy.de/sw/x86_64_gcc82_centos7/v02-02-02/DD4hep/v01-11-02/include/DDRec/DetectorData.h:43: undefined reference to `dd4hep::rec::operator<<(std::ostream&, dd4hep::rec::StructExtension<dd4hep::rec::FixedPadSizeTPCStruct> const&)'
collect2: error: ld returned 1 exit status
make[2]: *** [lib/libTester.so] Error 1
make[1]: *** [CMakeFiles/Tester.dir/all] Error 2
make: *** [all] Error 2
```
#### Expected
Compiles successfully

## Workaround
Explicitly link `DDRec` component in the `CMakeLists.txt` with
```
find_package(DD4hep REQUIRED COMPONENTS DDRec)
target_link_libraries(${PROJECT_NAME} ${DD4hep_COMPONENT_LIBRARIES})
```

## Affects
`source /cvmfs/ilc.desy.de/sw/x86_64_gcc82_centos7/v02-02-02/init_ilcsoft.sh`

### PR DESCRIPTION
**Pull request by dudarboh**
BEGINRELEASENOTES

- Added `DD4hep::DDRec` library as a dependency to avoid undefined reference symbol errors using `dd4hep::rec` features. Fixes #26 

ENDRELEASENOTES

Fix #26 . In case there is strong reason not to do that, which I am missing, we should do that. Otherwise `dd4hep::rec` features from the source file are unusable in the Marlin Processors, which are built with only Marlin and MarlinUtil as the dependencies.

### CODE REVIEW

**Review by tmadlener — APPROVED** [review]
Very nice, Thanks. Waiting for CI to finish and then merging.

**Comment by tmadlener** [comment]
Build errors are unrelated to this PR. The CI environment seems to pick up the wrong version of DD4hep. #38 adds a Key4hep based workflow, once thta is merged, I will rebase this onto that to make sure things work there.

**Comment by dudarboh** [comment]
I don't remember that this issue affected anything else except MarlinUtil to be honest... So I don't think there was a "fix" somwehere. But I might have forgotten...

**Pull request merged**
